internal/exec/vendor_utils.go

```diff
@@ -668,10 +669,66 @@ func validateURI(uri string) error {
 }
 func isValidScheme(scheme string) bool {
 	validSchemes := map[string]bool{
-		"http":  true,
-		"https": true,
-		"git":   true,
-		"ssh":   true,
+		"http":       true,
+		"https":      true,
+		"git":        true,
+		"ssh":        true,
+		"git::https": true,
 	}
 	return validSchemes[scheme]
 }
+
+// CustomGitHubDetector intercepts GitHub URLs and transforms them
+// into something like git::https://<token>@github.com/... so we can
+// do a git-based clone with a token.
+type CustomGitHubDetector struct{}
+
+// Detect implements the getter.Detector interface for go-getter v1.
+func (d *CustomGitHubDetector) Detect(src, _ string) (string, bool, error) {
+	if len(src) == 0 {
+		return "", false, nil
+	}
+	// If there's no scheme (no "://"), prepend "https://".
+	if !strings.Contains(src, "://") {
+		src = "https://" + src
+	}
+
+	u, err := url.Parse(src)
+	if err != nil {
+		return "", false, fmt.Errorf("failed to parse GitHub URL %q: %w", src, err)
+	}
+
+	if u.Host != "github.com" {
+		return "", false, nil
+	}
+
+	parts := strings.SplitN(u.Path, "/", 4)
+	if len(parts) < 3 {
+		return "", false, fmt.Errorf("invalid GitHub URL %q", u)
+	}
+
+	token := os.Getenv("GITHUB_TOKEN")
+	if token != "" {
+		user := u.User.Username()
+		pass, _ := u.User.Password()
+		if user == "" && pass == "" {
+			u.User = url.UserPassword("x-access-token", token)
+		}
+	}
```

Comment:
Can you please add some log debug statements so it's clear that the token is not injected when it wasn't found, or that the token was injected?

Also if we don't inject it because there is a username or password, let's log that too.